src/shared/orgShapeListUtils.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { OrgAuthorization, Org, Logger, Connection } from '@salesforce/core';
+
+interface OrgShape {
+  Id: string;
+  CreatedBy: {
+    Username: string;
+  };
+  Status: string;
+  CreatedDate: string;
+}
+
+export interface OrgShapeListResult {
+  orgId: string;
+  username: string;
+  alias?: string;
+  shapeId: string;
+  status: string;
+  createdBy: string;
+  createdDate: string;
+}
+
+export interface JsForceError extends Error {
+  errorCode: string;
+  fields: string[];
+}
+
+export async function getAllShapesFromOrg(orgAuth: OrgAuthorization): Promise<OrgShapeListResult[]> {
+  const org = await Org.create({ aliasOrUsername: orgAuth.username });
+  const conn = org.getConnection();
+  const logger = await Logger.child(`getAllShapesFromOrg, ${orgAuth.username}`);
+  logger.info(`Query org: ${orgAuth.username} for shapes`);
+  try {
+    const shapesFound = await conn.query<OrgShape>(
+      "SELECT Id, Status, CreatedBy.Username, CreatedDate FROM ShapeRepresentation WHERE Status IN ( 'Active', 'InProgress' )"
+    );
+    return shapesFound.records.map((shape) => ({
+      ...{
+        orgId: orgAuth.orgId,
+        username: orgAuth.username,
+        shapeId: shape.Id,
+        status: shape.Status,
+        createdBy: shape.CreatedBy.Username,
+        createdDate: shape.CreatedDate,
+      },
+      ...(orgAuth.aliases.length > 0 ? { alias: orgAuth.aliases.join(',') } : {}),
```

Comment:
Even if no aliases are found, `orgAuth.aliases` is always an array, if `length > 0` then there's alias(es) defined in the authfile.

See:
https://github.com/forcedotcom/sfdx-core/blob/3b3dc662cabe912b077f7513e20970903d745f4e/src/org/authInfo.ts#L290
https://github.com/forcedotcom/sfdx-core/blob/3b3dc662cabe912b077f7513e20970903d745f4e/src/globalInfo/accessors/aliasAccessor.ts#L32